documentation/src/main/asciidoc/topics/query.adoc

```diff
@@ -18,480 +18,570 @@ Apart from indexed queries, {brandname} can run queries over non-indexed or
 
-=== Quick example
+NOTE: When using the full-text operator (:) or any other full-text feature, indexing is mandatory. {brandname} will
+throw an error if a full-text query is attempted in a non-indexed cache.
 
-We're going to store _Book_ instances in an {brandname} cache called "books". _Book_ instances will be indexed, so we enable indexing for the cache,
-letting {brandname} link:#query_autoconfig[configure the indexing automatically]:
+=== Configuration
 
-{brandname} configuration:
+To enable indexing via XML, you need to add the `<indexing>` element to your cache configuration, specify the entities that are indexed and optionally pass additional properties.
+
+NOTE: The presence of an `<indexing>` element which omits the `enabled` attribute will auto-enable indexing for your
+convenience, even though the default value of the `enabled` attribute is defined as `"false"` in the XSD schema.
+This behaviour exists only on schemas starting with version 11.
 
-.infinispan.xml
 [source,xml,options="nowrap",subs=attributes+]
 ----
-include::config_examples/infinispan_distributed_cache_books.xml[]
+include::config_examples/indexing.xml[]
 ----
 
-Obtaining the cache:
+Programmatic:
 
 [source,java]
 ----
-import org.infinispan.Cache;
-import org.infinispan.manager.DefaultCacheManager;
-import org.infinispan.manager.EmbeddedCacheManager;
-
-EmbeddedCacheManager manager = new DefaultCacheManager("infinispan.xml");
-Cache<String, Book> cache = manager.getCache("books");
+import org.infinispan.configuration.cache.*;
 
+ConfigurationBuilder cacheCfg = ...
+cacheCfg.indexing().enable()
+            .addIndexedEntity(Book.class)
+      .addProperty("property name", "propery value")
 ----
 
-Each _Book_ will be defined as in the following example; we have to choose which properties are indexed, and for each property we can optionally choose advanced indexing options using the annotations defined in the Hibernate Search project.
-
-[source,java]
-.Book.java
-----
-import org.hibernate.search.annotations.*;
-import java.util.Date;
-import java.util.HashSet;
-import java.util.Set;
+NOTE: When using programmatic configuration, indexing must be enabled explicitly by invoking the `enable()` method, as
+no auto-enabling mechanism exist for programmatic configuration as in the case of XML configuration.
 
-//Values you want to index need to be annotated with @Indexed, then you pick which fields and how they are to be indexed:
-@Indexed
-public class Book {
-   @Field String title;
-   @Field String description;
-   @Field @DateBridge(resolution=Resolution.YEAR) Date publicationYear;
-   @IndexedEmbedded Set<Author> authors = new HashSet<Author>();
-}
+=== Specifying indexed Entities
 
-----
+It is recommended to declare the indexed types, as they will be mandatory in the next {brandname} version.
+This can be done via xml:
```

Comment:
I wonder if we could say "You should declare indexed types because {brandname} ..." then briefly explain why they will be mandatory in the next version.

It seems like we're adding a bit of maintenance overhead in the docs here too. This statement will need to be updated for 12 so might as well try to make the reason clear now.

Reply:
> why they will be mandatory in the next version.

Because we're moving away from it not being mandatory :)  
I would not worry about maintenance overhead of the query docs, it will suffer major changes in ISPN 12 as well